src/py/pyodide/utils/abort/polyfill.js

```diff
@@ -0,0 +1,52 @@
+function polyfillAbortSignalAny() {
+  /** @param {AbortSignal[]} signals */
+  return (signals) => {
+    if (AbortSignal.any) {
+      return AbortSignal.any(signals);
+    }
+    const controller = new AbortController();
+    const controllerRef = new WeakRef(controller);
+    /** @type {[WeakRef<AbortSignal>, (() => void)][]} */
+    const eventListenerPairs = [];
+    let followingCount = signals.length;
+
+    /** @type {FinalizationRegistry<(callback: () => any) => void>} */
+    const registry = (globalThis.__abortSignalCleanups =
+      globalThis.__abortSignalCleanups ??
+      new FinalizationRegistry((callback) => void callback()));
```

Comment:
Let's move this to the top of the file as:
```js
const registry = new FinalizationRegistry((callback) => void callback());
```

Reply:
I guess we should also do the check at top level:
```js
export let abortSignalAny;
if (AbortSignal.any) {
    abortSignalAny = AbortSignal.any;
} else {
    const registry = new FinalizationRegistry((callback) => void callback());
    // ... define abortSignalAny ponyfill
}
```

Reply:
Should we export it?

Reply:
Yeah, then import from `api.ts` and set `API.abortSignalAny = abortSignalAny`. Then from Python you can do 
```py
from pyodide_js._api import abortSignalAny
```
and then you can use it.

Reply:
I attached it to API and it should work now. I mean, now I did't `export let` it because the only usage is through `pyodide_js._api`. Do you think we need to add some test for it?

Reply:
Even the core tests surprisingly passed 😂